### ISSUE TITLE
oc:undeploy goal does not delete BuildConfig created by previous oc:resources goal executions

### ISSUE THREAD

**Issue by rmarting** [eclipse/jkube#332]
### Description
After deploy my application with the following command successfully in OCP several times (at least three times):

```mvn oc:resource oc:build oc:apply```

If I undeployed the application with ```oc:undeploy``` goal and I repeat the same deployment command, the following error appears:

```
❯ mvn oc:resource oc:build oc:apply
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------< com.jromanmartin.kafka:kafka-clients-sb-sample >-----------
[INFO] Building kafka-clients-sb-sample 2.5.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- openshift-maven-plugin:1.0.0-rc-1:resource (default-cli) @ kafka-clients-sb-sample ---
[INFO] oc: Using docker image name of namespace: amq-streams
[INFO] oc: Running generator spring-boot
[INFO] oc: spring-boot: Using Docker image quay.io/jkube/jkube-java-binary-s2i:0.0.7 as base / builder
[INFO] oc: using resource templates from /home/rmarting/Workspaces/ws-streams/kafka-clients-sb-sample/src/main/jkube
[INFO] oc: jkube-service: Adding a default service 'kafka-clients-sb-sample' with ports [8080]
[INFO] oc: jkube-revision-history: Adding revision history limit to 2
[INFO] oc: validating /home/rmarting/Workspaces/ws-streams/kafka-clients-sb-sample/target/classes/META-INF/jkube/openshift/kafka-clients-sb-sample-service.yml resource
[INFO] oc: validating /home/rmarting/Workspaces/ws-streams/kafka-clients-sb-sample/target/classes/META-INF/jkube/openshift/kafka-clients-sb-sample-deploymentconfig.yml resource
[INFO] oc: validating /home/rmarting/Workspaces/ws-streams/kafka-clients-sb-sample/target/classes/META-INF/jkube/openshift/kafka-clients-sb-sample-route.yml resource
[INFO] 
[INFO] --- openshift-maven-plugin:1.0.0-rc-1:build (default-cli) @ kafka-clients-sb-sample ---
[INFO] oc: Using OpenShift build with strategy S2I
[INFO] oc: Running in OpenShift mode
[INFO] oc: Running generator spring-boot
[INFO] oc: spring-boot: Using Docker image quay.io/jkube/jkube-java-binary-s2i:0.0.7 as base / builder
[INFO] oc: [kafka-clients-sb-sample:latest] "spring-boot": Created docker source tar /home/rmarting/Workspaces/ws-streams/kafka-clients-sb-sample/target/docker/kafka-clients-sb-sample/latest/tmp/docker-build.tar
[INFO] oc: Creating BuildServiceConfig kafka-clients-sb-sample-s2i for Source build
[INFO] oc: Adding to ImageStream kafka-clients-sb-sample
[INFO] oc: Starting Build kafka-clients-sb-sample-s2i
[ERROR] oc: Build for kafka-clients-sb-sample-s2i failed: stream was reset: NO_ERROR
[ERROR] oc: kafka-clients-sb-sample-s2i-1		BuildPodExists	The pod for this build already exists and is older than the build.
[ERROR] oc: Failed to execute the build [Unable to build the image using the OpenShift build service]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  9.607 s
[INFO] Finished at: 2020-07-31T13:11:41+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.0.0-rc-1:build (default-cli) on project kafka-clients-sb-sample: Failed to execute the build: Unable to build the image using the OpenShift build service: BuildPodExists The pod for this build already exists and is older than the build. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

### Info

* Eclipse JKube version : ```1.0.0-rc-1```
* Maven version (`mvn -v`) : 
```
Apache Maven 3.6.1 (Red Hat 3.6.1-5)
Maven home: /usr/share/maven
Java version: 11.0.8, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-11-openjdk-11.0.8.10-2.fc32.x86_64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.7.10-201.fc32.x86_64", arch: "amd64", family: "unix"
```
* Red Hat OpenShift version : ```4.5.4```

* If it's a bug, how to reproduce : Basically execute the following command sequence:

```
$ mvn oc:resource oc:build oc:apply
$ mvn oc:undeploy
$ mvn oc:resource oc:build oc:apply
```

* Sample Reproducer Project : https://github.com/rmarting/kafka-clients-sb-sample

**Comment by rohanKanojia** [comment]
I think `ImageStreams` are also not cleaned up during `oc:undeploy`

**Comment by rohanKanojia** [comment]
these `BuildConfig` and `ImageStreams` are not created during `oc:resource` phase.They are created during `oc:build` phase in OpenShiftBuildService. Maybe that's why jkube can't find them in `target/classes/META-INF/jkube/openshift.yml` and why they are not deleted...

**Comment by manusa** [comment]
> I think ImageStreams are also not cleaned up during oc:undeploy

This was addressed a while back in #170. But probably some stuff was left out since this was not the main purpose of the PR.
We need to revisit.
